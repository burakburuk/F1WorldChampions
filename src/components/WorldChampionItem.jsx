import React, { PureComponent } from 'react';
import Skeleton from 'react-loading-skeleton';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Button from './Button';
import { TableRow, TableCell } from './Table';

const StyledYear = styled.div`
    height: 100%;
    text-align: center;
    background: #bfbfbf;
    color: white;
    padding-top: 16px;
    margin-right:5px;
    padding-left: 4px;
    padding-right: 4px;
    border-right: 5px solid red;
`;

const StyledTeam = styled.div`
    color: gray;
    text-align: center;
    padding-top: 2px;
    padding-right: 10px;
`;

const StyledCompany = styled.div`
    float: left;
    padding: 5px;
`;

const StyledImage = styled.img`
    width: 93px;
    text-align: center;
    padding-top: 0;
    float: right;
`;

const StyledButton = styled(Button)`
    width: 100%;
`;

class WorldChampionItem extends PureComponent {
    render() {
        const {
            name,
            year,
            nationality,
            company,
            points,
            carImage,
        } = this.props;

        return (
            <TableRow>
                <TableCell width={70}>
                    <StyledYear>
                        {year || <Skeleton />}
                    </StyledYear>
                </TableCell>
                <TableCell>
                    {name || <Skeleton />}
                </TableCell>
                <TableCell>
                    {nationality || <Skeleton />}
                </TableCell>
                <TableCell>
                    <StyledTeam>
                        {company ? <StyledCompany>{company}</StyledCompany> : <Skeleton />}
                        {company && <StyledImage alt="no_image" src={carImage}/>}
                    </StyledTeam>
                </TableCell>
                <TableCell width={52}>
                    {points || <Skeleton />}
                </TableCell>
                <TableCell width={140} align={'right'}>
                    <StyledButton
                        primary={!!name}
                        onClick={this.handleOpenChampionsByYearPopup}>
                        Show details
                    </StyledButton>
                </TableCell>
            </TableRow>
        );
    }

    handleOpenChampionsByYearPopup = () => {
        const { year, driverId } = this.props;
        if (year && driverId) {
            this.props.openChampionsByYearPopup(year, driverId);
        }
    }
}

WorldChampionItem.propTypes = {
    driverId: PropTypes.string,
    name: PropTypes.string,
    year: PropTypes.number,
    nationality: PropTypes.string,
    company: PropTypes.string,
    points: PropTypes.number,
    carImage: PropTypes.string,
    openChampionsByYearPopup: PropTypes.func,
};

export default WorldChampionItem;
