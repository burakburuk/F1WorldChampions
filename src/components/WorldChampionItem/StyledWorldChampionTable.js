import styled from 'styled-components';
import Button from '../Button';

const StyledYear = styled.div`
height: 100%;
text-align: center;
background: #bfbfbf;
color: white;
padding-top: 16px;
margin-right:5px;
padding-left: 4px;
padding-right: 4px;
border-right: 5px solid red;
`;

const StyledTeam = styled.div`
color: gray;
text-align: center;
padding-top: 2px;
padding-right: 10px;
`;

const StyledCompany = styled.div`
float: left;
padding: 5px;
`;

const StyledImage = styled.img`
width: 93px;
text-align: center;
padding-top: 0;
float: right;
`;

const StyledButton = styled(Button)`
width: 100%;
`;

export {
    StyledYear,
    StyledTeam,
    StyledCompany,
    StyledImage,
    StyledButton,
};
