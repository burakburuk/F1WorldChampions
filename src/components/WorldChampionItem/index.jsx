import React, { PureComponent } from 'react';
import Skeleton from 'react-loading-skeleton';
import PropTypes from 'prop-types';
import { TableRow, TableCell } from '../Table';

import {
    StyledYear,
    StyledTeam,
    StyledCompany,
    StyledImage,
    StyledButton,
} from './StyledWorldChampionTable';

class WorldChampionItem extends PureComponent {
    render() {
        const {
            name,
            year,
            nationality,
            company,
            points,
            carImage,
        } = this.props;

        return (
            <TableRow>
                <TableCell width={70}>
                    <StyledYear>
                        {year || <Skeleton />}
                    </StyledYear>
                </TableCell>
                <TableCell>
                    {name || <Skeleton />}
                </TableCell>
                <TableCell>
                    {nationality || <Skeleton />}
                </TableCell>
                <TableCell>
                    <StyledTeam>
                        {company ? <StyledCompany>{company}</StyledCompany> : <Skeleton />}
                        {company && <StyledImage alt="no_image" src={carImage}/>}
                    </StyledTeam>
                </TableCell>
                <TableCell width={52}>
                    {points || <Skeleton />}
                </TableCell>
                <TableCell width={140} align={'right'}>
                    <StyledButton
                        primary={!!name}
                        onClick={this.handleOpenChampionsByYearPopup}>
                        Show details
                    </StyledButton>
                </TableCell>
            </TableRow>
        );
    }

    handleOpenChampionsByYearPopup = () => {
        const { year, driverId } = this.props;
        if (year && driverId) {
            this.props.openChampionsByYearPopup(year, driverId);
        }
    }
}

WorldChampionItem.propTypes = {
    driverId: PropTypes.string,
    name: PropTypes.string,
    year: PropTypes.number,
    nationality: PropTypes.string,
    company: PropTypes.string,
    points: PropTypes.number,
    carImage: PropTypes.string,
    openChampionsByYearPopup: PropTypes.func,
};

export default WorldChampionItem;
