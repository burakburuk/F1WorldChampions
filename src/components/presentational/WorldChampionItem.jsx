import React, { PureComponent } from 'react';
import Skeleton from 'react-loading-skeleton';
import PropTypes from 'prop-types';
import Button from './Button';

class WorldChampionItem extends PureComponent {
    render() {
        const {
            name,
            year,
            nationality,
            company,
            points,
            carImage,
        } = this.props;

        return (
            <tr>
                <td>
                    <div className="ch-item item-year">
                        {year || <Skeleton />}
                    </div>
                </td>
                <td>
                    <div className="ch-item item-driver">
                        {name || <Skeleton />}
                    </div>
                </td>
                <td>
                    <div className="ch-item">
                        {nationality || <Skeleton />}
                    </div>
                </td>
                <td>
                    <div className="ch-item">
                        {company ? <div className="item-company">{company}</div> : <Skeleton />}
                        {company && <img className="item-image" alt="no_image" src={carImage}/>}
                    </div>
                </td>
                <td>
                    <div className="ch-item item-points">
                        {points || <Skeleton />}
                    </div>
                </td>
                <td>
                    <div className="ch-item">
                        <Button
                            color={name ? 'primary' : 'disabled'}
                            className="show-details"
                            onClick={this.handleOpenChampionsByYearPopup}>
                            Show details
                        </Button>
                    </div>
                </td>
            </tr>
        );
    }

    handleOpenChampionsByYearPopup = () => {
        const { year, driverId } = this.props;
        if (year && driverId) {
            this.props.openChampionsByYearPopup(year, driverId);
        }
    }
}

WorldChampionItem.propTypes = {
    driverId: PropTypes.string,
    name: PropTypes.string,
    year: PropTypes.number,
    nationality: PropTypes.string,
    company: PropTypes.string,
    points: PropTypes.number,
    carImage: PropTypes.string,
    openChampionsByYearPopup: PropTypes.func,
};

export default WorldChampionItem;
