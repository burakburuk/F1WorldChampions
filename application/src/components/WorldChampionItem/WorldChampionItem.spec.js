import React from 'react';
import { shallow } from 'enzyme';
import WorldChampionItem from '.';

it('should match with snapshot', () => {
    const wrapper = shallow(<WorldChampionItem />);
    expect(wrapper).toMatchSnapshot();
});
