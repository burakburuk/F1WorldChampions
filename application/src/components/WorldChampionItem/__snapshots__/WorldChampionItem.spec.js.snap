// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<TableRow>
  <TableCell
    width={70}
  >
    <styled.div>
      <Skeleton
        count={1}
        duration={1.2}
        width={null}
        wrapper={null}
      />
    </styled.div>
  </TableCell>
  <TableCell>
    <Skeleton
      count={1}
      duration={1.2}
      width={null}
      wrapper={null}
    />
  </TableCell>
  <TableCell>
    <Skeleton
      count={1}
      duration={1.2}
      width={null}
      wrapper={null}
    />
  </TableCell>
  <TableCell>
    <styled.div>
      <Skeleton
        count={1}
        duration={1.2}
        width={null}
        wrapper={null}
      />
    </styled.div>
  </TableCell>
  <TableCell
    width={52}
  >
    <Skeleton
      count={1}
      duration={1.2}
      width={null}
      wrapper={null}
    />
  </TableCell>
  <TableCell
    align="right"
    width={140}
  >
    <Styled(Button)
      onClick={[Function]}
      primary={false}
    >
      Show details
    </Styled(Button)>
  </TableCell>
</TableRow>
`;
